src/handler.js

```diff
@@ -97,99 +76,30 @@ const handle = async message => {
-    const s3 = new AWS.S3();
-
-    s3.upload({
-      Bucket: process.env.S3_BUCKET,
-      Key: emojiAlias,
-      // Body: writeStream,
-      Body: fs.createReadStream(tmpPath),
-      ContentType: metadata.file.mimetype,
-      ACL: 'public-read',
-    })
-      .promise()
-      .then(async response => {
-        logger.info(`Uploaded to s3 ${response.Location}`);
-        const b64 = fs.readFileSync(tmpPath, { encoding: 'base64' });
-
-        slackAsPromise('chat.postMessage', {
-          channel: message.channel_id,
-          text: 'This is my attempt at the emoji you asked for',
-          attachments: JSON.stringify([
-            { fallback: tmp, image_url: response.Location },
-          ]),
-        })
-          .then(() => {
-            logger.info(`Sent to slack: ${tmp}`);
-          })
-          .catch(e => {
-            logger.error(`Failed to upload to slack: ${tmp}`, e);
-          });
-
-        octokit.authenticate({
-          type: 'token',
-          token: process.env.GITHUB_TOKEN,
-        });
-
-        // allows full url or just owner/repo
-        const [repo, owner] = process.env.GITHUB_REPO.split('/').reverse();
-        const branch = process.env.GITHUB_REPO_BRANCH;
-        const emojiRepoDir = process.env.GITHUB_REPO_DIR;
-        const emojiPath = path.join(
-          emojiRepoDir,
-          `${emojiAlias}.${metadata.file.filetype}`
-        );
-
-        try {
-          await octokit.repos.createFile({
-            owner,
-            repo,
-            path: emojiPath,
-            branch,
-            message: `emojo: added :${emojiAlias}:`,
-            content: b64,
-          });
-
-          logger.info(
-            `Created ${emojiAlias}.${
-              metadata.file.filetype
-            } in ${owner}/${repo}`
-          );
-        } catch (e) {
-          logger.error(
-            `Error creating file on Github ${owner}/${repo}/${emojiPath}`,
-            e
-          );
-          // @todo: this needs handling as a proper response
-        }
-      })
-      .catch(e => {
-        logger.error(`Failed to upload to s3: ${tmp} %s`, e);
-      });
-
-    return {
-      done: true,
-    };
-  });
+  const resizer = getResizer(metadata.file.mimetype);
+  const image = await downloadAsStream(metadata.file.url_private);
+
+  // We can add the other stream here
+  image.data.pipe(resizer).pipe(writeStream);
+
+  // We can add the other stream here when we have it w Promise.all
+  await streamAsPromise(writeStream);
+
+  // Both of these currently rely on the write to file happening which is why we
+  // need to wait for that at the moment. What we can do is pass the stream into
+  // uploadToGithub and sendPreview and that way we can just have them only
+  // resolve when their respective stream has finished and remove the bit above
+  await Promise.all([
+    sendPreview(emojiAlias, metadata, tmpPath),
+    uploadToGithub(emojiAlias, metadata, tmpPath),
+  ]);
```

Comment:
We should catch here

Reply:
Uncaught exceptions go back to the lambda handler as an error with the exception message as the string. I'd intentionally left a few of these for the moment until we standardise how the error handling works.

One option is to wrap the whole lot in a try/catch so that we can put a uniform json error in one place. bit like express does while offering an exception handler route kinda thing if that makes sense.

Anyway, could catch, but at present other than offering a place to log or slack about the problem etc it won't change much.

I vote standardise error handling in another PR